src/box/lua/schema.lua

```diff
@@ -3391,7 +3392,9 @@ end
 local function chpasswd(uid, new_password)
     local _user = box.space[box.schema.USER_ID]
     check_password(new_password)
-    _user:update({uid}, {{"=", 5, prepare_auth_list(new_password)}})
+    _user:update({uid}, {{'=', 5, prepare_auth_list(new_password)},
+                         {'=', 6, {}},
```

Comment:
It seems that clearing history is exactly what this function must not do.

Reply:
This commit just adds an empty history field. The history is enabled in the last commit of this series:

<details><summary>diff</summary>

```diff
diff --git a/src/box/lua/schema.lua b/src/box/lua/schema.lua
index cd06564e7b2d..52161681818c 100644
--- a/src/box/lua/schema.lua
+++ b/src/box/lua/schema.lua
@@ -3383,17 +3383,26 @@ local function prepare_auth_list(password)
     }
 end
 
-local function check_password(password)
+local function prepare_auth_history(uid)
+    if internal.prepare_auth_history ~= nil then
+        return internal.prepare_auth_history(uid)
+    else
+        return {}
+    end
+end
+
+local function check_password(password, auth_history)
     if internal.check_password ~= nil then
-        internal.check_password(password)
+        internal.check_password(password, auth_history)
     end
 end
 
 local function chpasswd(uid, new_password)
     local _user = box.space[box.schema.USER_ID]
-    check_password(new_password)
+    local auth_history = prepare_auth_history(uid)
+    check_password(new_password, auth_history)
     _user:update({uid}, {{'=', 5, prepare_auth_list(new_password)},
-                         {'=', 6, {}},
+                         {'=', 6, auth_history},
                          {'=', 7, math.floor(fiber.time())}})
 end
 
```

</details>